src/utils/transcript.ts

```diff
@@ -0,0 +1,114 @@
+export interface Transcript {
+	index: number;
+	start: number;
+	end: number;
+}
+
+enum FIND_PART_STRATEGY {
+	NEXT = 'NEXT',
+	PREV = 'PREV',
+	CLOSEST = 'CLOSEST',
+}
+// Use binary search to find the word in transcript with [start, end]
+// containing the given timestamp.
+// @param strategy {string}, one of ['NEXT', 'PREV', 'CLOSEST'], the fallback
+// strategy for handling timestamps that do not occur within a spoken part
+// - NEXT: for bounding the start of a highlight range
+// - PREV: for bounding the end of a highlight range
+// - CLOSEST: for any other use
+function getMatchingPartToTimestamp(
```

Comment:
Should this be part of the video-player library itself (`src/utils`) - or only part of the storybook code 
(`.storybook`)? Making it part of the library itself also means that we will have to maintain and keep API stability.

Reply:
Let's keep it for the moment only for storybook.
If we would add a hook "useKaraoke", that will handle all "karaoke" functionality  + search algotirtms and so on - then yes. But now this is only a "storybook thing".
So better to move it in `.storybook` "workspace" (.storybook/stories/shared/)